### ISSUE TITLE
Absolute Max Pooling 

### ISSUE THREAD

**Issue by ningerlove** [KaiyuYue/mgd#5]
def amp(self, i, t_feats, s_feats, margins):
        """
        Absolute Max Pooling for channels reduction.
        """
        b, sc, h, w = s_feats[i].shape
        _, tc, _, _ = t_feats[i].shape

        groups = tc // sc

        t = []
        m = []
        for c in range(0, tc, sc):
            if c == (tc // sc) * sc and self.shave:
                continue

            t.append(t_feats[i][:, self.guided_inds[i][c:c+sc].detach(), :, :])
            m.append(margins[:, self.guided_inds[i][c:c+sc].detach(), :, :])

        t = torch.stack(t, dim=2)
        m = torch.stack(m, dim=2)

        t = t.reshape(b, sc, groups, -1)
        m = m.reshape(1, sc, groups, -1)

        t_inds = torch.argmax(t, dim=2)

        t = t.gather(2, t_inds.unsqueeze(2))
        m = m.mean(dim=2)

        t = t.reshape(b, sc, h, w)
        m = m.reshape(1, sc, 1, 1)
for Absolute Max Pooling ,t may be abs ?
     the line  t_inds = torch.argmax(t, dim=2) should change to t_inds = torch.argmax(torch.abs(t), dim=2)

**Comment by kaiyuyue** [comment]
Ops, my bad. You're right. I have fixed it. If you have further experimental problems, feel free to post them here. Thanks very much.
